cou/utils/juju_utils.py

```diff
@@ -168,27 +168,27 @@ def name(self) -> str:
         return self._name
 
     @retry(no_retry_exceptions=(ActionFailed,))
-    async def _get_action_result(self, action: Action, raise_on_failure: bool) -> Action:
-        """Get results from action.
+    async def _get_waited_action_object(self, action: Action, raise_on_failure: bool) -> Action:
```

Comment:
not-blocking: In output, we are saying that it's awaited, so why not also `_get_awaited_action_object`? IMO changing the name is not necessary.

Reply:
`unit.run_action` is also a sync command with `awaits`, so I think `_get_awaited_action_object` would a misleading name. I would prefer "waited" because it indicates the usage of `action.wait()` function.

Also, since the function is not returning just the action results but a action object with results as part of it, I renamed the function for more clarity.